### ISSUE TITLE
Accessibility Concerns with New Badge Design Language

### ISSUE THREAD

**Issue by CodeRunner42** [PrairieLearn/PrairieLearn#14317]
### Details

As a student currently using PL, I'd like to raise concerns regarding accessibility needs after the recent merge of updated icon styling.

The low saturation, paired with the colouring between the lettering and the icons greatly reduces readability for badges. At a glance legibility has been greatly affected compared to the original implementation of the badges.

I've provided an example where previously clear differences between the badge colours have now been drastically reduced, this is most evident on the badges for A/PA and Q/PQ respectively. 

<img width="80" height="462" alt="Image" src="https://github.com/user-attachments/assets/ed76dbbd-6413-443e-a497-d9b9690bad0e" />

<img width="60" height="75" alt="Image" src="https://github.com/user-attachments/assets/e0e7468b-4617-4eea-85d6-1c712c636e65" />

**Comment by nwalters512** [comment]
Thanks for the feedback! Can you share which institution/course this is in so I can check out which colors they've selected for their assessment sets? Also, where did the smaller image with "Q" and "PQ" come from?

**Comment by CodeRunner42** [comment]
Appreciate the follow-up. The screenshots are from the CPSC 213 course at UBC. The smaller image comes from a grade breakdown that students receive, but it was mainly used for illustrative purposes as I don't have any images of the original badges on hand. It gets as close to the original badges as I remember, but they are different in colour, even prior to the updated badges.

**Comment by nwalters512** [comment]
Thanks, that's helpful!

I played around with tweaking some things here - do these look better to you?

<img width="1153" height="791" alt="Image" src="https://github.com/user-attachments/assets/8aeac920-ad9f-46fe-9d0a-8132805277db" />

**Comment by CodeRunner42** [comment]
The tweaks offer much better contrast between badge colour, but the text legibility is still a bit lacking. Especially for 3rd colour variants, the backgrounds and text almost blend together with it not being helped on the white background.

I've read the original PR #14295 and I understand the intention to bring the design language to be in line with Github, but as PL is restricted to a lighter theme, this creates a greater challenge in balancing the badge legibility with how colour and contrast is used.  

Saturating the backgrounds helps with telling apart between the variants of each colour, but the text itself gets harder to read, especially for the darker tones. The most evident examples would be of Grey 3 and Blue 3 respectively.

I appreciate your continued reception to feedback.

**Comment by nwalters512** [comment]
Under the proposed changes, `gray3` and `blue3` actually have some of the highest contrast out of any of them at `8.26:1` and `8.11:1`, respectively.

The changes are now up in #14327. These are definitely an improvement, so we'll land them. If you have concrete suggestions on how you might increase legibility, I'd encourage you to open a PR once these refinements land!

I'll note that the goal was not to be "in line" with GitHub - it just used their badge style as inspiration. We're ultimately taking a different approach then they are. The real goal was to make the badges easier on the eyes when lots of different ones are used together, as they would be on instructor facing pages like the questions table and assessment questions editor. These aren't pages that students ever see.
